document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoSelectingExpressionParser.java

```diff
@@ -1,74 +1,99 @@
 package org.hypertrace.core.documentstore.mongo.parser;
 
-import static java.util.function.Predicate.not;
+import static java.util.stream.Collectors.toMap;
+import static org.hypertrace.core.documentstore.mongo.MongoUtils.getUnsupportedOperationException;
 
 import com.mongodb.BasicDBObject;
-import com.mongodb.client.model.Projections;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
-import org.apache.commons.collections4.CollectionUtils;
 import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
 import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
 import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
 import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
-import org.hypertrace.core.documentstore.parser.SelectingExpressionParser;
-import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.parser.SelectingExpressionVisitor;
+import org.hypertrace.core.documentstore.query.QueryInternal;
 import org.hypertrace.core.documentstore.query.SelectionSpec;
 
-public class MongoSelectingExpressionParser extends MongoExpressionParser
-    implements SelectingExpressionParser {
+public abstract class MongoSelectingExpressionParser implements SelectingExpressionVisitor {
 
   private static final String PROJECT_CLAUSE = "$project";
 
-  private final String identifierPrefix;
+  protected final MongoSelectingExpressionParser baseParser;
 
-  public MongoSelectingExpressionParser(final Query query) {
-    this(query, false);
+  protected MongoSelectingExpressionParser() {
+    this(null);
```

Comment:
Is it possible to create a default no-op selection parser?

Reply:
Done.